client/src/TenderList.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+
+const TendersList = () => {
+	const [tenders, setTenders] = useState([]);
+	const [loading, setLoading] = useState(true);
+	const [page, setPage] = useState(1);
+	const [hasMore, setHasMore] = useState(true);
+	const [error, setError] = useState(null);
+
+	const fetchTenders = async (pageNumber) => {
+		setLoading(true);
+		try {
+			const response = await fetch(`/api/tenders?page=${pageNumber}`);
+			if (!response.ok) {
+				throw new Error(`HTTP error! status: ${response.status}`);
+			}
+			const data = await response.json();
+
+			console.log("Fetched data:", data);
+
+			if (Array.isArray(data.tenders)) {
+				setTenders((prevTenders) => [...prevTenders, ...data.tenders]);
+				setHasMore(data.tenders.length === 25);
+				setError(null);
+			} else {
+				throw new Error("Invalid data format: data.tenders is not an array");
+			}
+		} catch (error) {
+			setError("Error fetching tenders: " + error.message);
+		} finally {
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		fetchTenders(page);
+	}, [page]);
+
+	const loadMore = () => {
+		if (hasMore && !loading) {
+			setPage((prevPage) => prevPage + 1);
+		}
+	};
+
+	return (
+		<div>
+			<h2>Tenders List</h2>
+			{error && <p style={{ color: "red" }}>{error}</p>}
+			<table>
+				<thead>
+					<tr>
+						<th>Tender ID</th>
+						<th>Tender Title</th>
+						<th>Tender Created Date</th>
+						<th>Tender Announcement Date</th>
+						<th>Tender Closing Date</th>
+						<th>Tender Project Deadline Date</th>
+						<th>Tender Status</th>
+					</tr>
+				</thead>
+				<tbody>
+					{console.log(tenders)}
```

Comment:
Remove all `console.log`s from your code

Reply:
I have done that